code/controllers/configuration/configuration.dm

```diff
@@ -123,11 +124,26 @@
 		if(lockthis)
 			E.protection |= CONFIG_ENTRY_LOCKED
 
+		if(E.deprecated_by)
+			var/datum/config_entry/new_ver = entries_by_type[E.deprecated_by]
+			var/new_value = E.DeprecationUpdate(value)
+			var/good_update = istext(new_value)
+			log_config("Entry [entry] is deprecated and will be removed soon. Migrate to [new_ver.name]![good_update ? " Suggested new value is: [new_value]" : ""]")
+			if(!warned_deprecated_configs)
+				addtimer(CALLBACK(GLOBAL_PROC, /proc/message_admins, "This server is using deprecated configuration settings. Please check the logs and update accordingly."), 0)
+				warned_deprecated_configs = TRUE
+			if(good_update)
+				value = new_value
+				E = new_ver
+			else
+				warning("[new_ver.type] is deprecated but gave no proper return for DeprecationUpdate()")
+			
 		var/validated = E.ValidateAndSet(value)
 		if(!validated)
 			log_config("Failed to validate setting \"[value]\" for [entry]")
-		else if(E.modified && !E.dupes_allowed)
-			log_config("Duplicate setting for [entry] ([value], [E.resident_file]) detected! Using latest.")
+		else 
+			if(E.modified && !E.dupes_allowed)
+				log_config("Duplicate setting for [entry] ([value], [E.resident_file]) detected! Using latest.")

```

Comment:
As an unrelated aside this needs to go.

All configs should be designed such that they can be overridden to allow for the synced configs model /tg/ is using.